File: tools/session-manager/fixtures/typescript/get-state.ts
#!/usr/bin/env node
import { spawnSync } from "node:child_process";

const skillName: string = String(process.argv[2] ?? "");
if (!skillName) process.exit(2);

const result = spawnSync("session-manager", ["get-state", "--skill-name", skillName], {
  encoding: "utf8",
});
process.stdout.write(result.stdout ?? "");
process.stderr.write(result.stderr ?? "");
process.exit(result.status ?? 1);

File: tools/session-manager/fixtures/python/get_state.py
#!/usr/bin/env python3
import subprocess
import sys


def main() -> int:
    if len(sys.argv) != 2:
        return 2
    result = subprocess.run(
        ["session-manager", "get-state", "--platform", sys.argv[1]],
        check=False,
        capture_output=True,
        text=True,
    )
    sys.stdout.write(result.stdout)
    sys.stderr.write(result.stderr)
    return result.returncode


if __name__ == "__main__":
    raise SystemExit(main())
